parquet-column/src/main/java/parquet/column/Dictionary.java

```diff
@@ -29,10 +30,15 @@
 public abstract class Dictionary {
 
   private final Encoding encoding;
+  private final PrimitiveType.PrimitiveTypeName primitiveTypeName;
 
-  public Dictionary(Encoding encoding) {
+  public Dictionary(PrimitiveType.PrimitiveTypeName primitiveTypeName, Encoding encoding) {
+    this.primitiveTypeName = primitiveTypeName;
     this.encoding = encoding;
   }
+  public PrimitiveType.PrimitiveTypeName getPrimitiveTypeName() {
```

Comment:
Thanks for the review, I think that the suggestion is good. It would mean creating a couple of new dictionary classes e.g. PlainInt96Dictionary. 

Are you happy with Dictionary carrying the type information, or at least having abstract method to get information. I did this to make adding dictionary support to SimplePrimitiveConverter simpler, and I added dictionary support to SimplePrimitiveConverter so I could test using existing framework.